src/rgw/rgw_metadata.h

```diff
@@ -70,7 +70,8 @@ class RGWMetadataHandler {
   virtual RGWMetadataObject *get_meta_obj(JSONObj *jo, const obj_version& objv, const ceph::real_time& mtime) = 0;
 
   virtual int get(string& entry, RGWMetadataObject **obj, optional_yield) = 0;
-  virtual int put(string& entry, RGWMetadataObject *obj, RGWObjVersionTracker& objv_tracker, optional_yield, RGWMDLogSyncType type) = 0;
+  virtual int put(string& entry, RGWMetadataObject *obj, RGWObjVersionTracker& objv_tracker, optional_yield,
+  RGWMDLogSyncType type, bool make_private = false) = 0;
```

Comment:
it's best to avoid arguments with default values in virtual functions, since the compiler doesn't check whether all overloads use the same default values. that can be a source of confusion and lead to bugs

if you need to line wrap function parameters, please indent the next line past the `(`. see https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions

the name `make_private` may be confusing without extra context. what do you think about calling this `from_remote_zone`?

Reply:
done.